Mathlib/Algebra/Group/AddChar.lean

```diff
@@ -85,49 +87,66 @@ open Multiplicative
 /-- Define coercion to a function so that it includes the move from `A` to `Multiplicative A`.
 After we have proved the API lemmas below, we don't need to worry about writing `ofAdd a`
 when we want to apply an additive character. -/
-instance instFunLike : FunLike (AddChar A M) A M :=
+instance instFunLike [Monoid M] : FunLike (AddChar A M) A M :=
   inferInstanceAs (FunLike (Multiplicative A →* M) A M)
 #noalign add_char.has_coe_to_fun
 
-theorem coe_to_fun_apply (ψ : AddChar A M) (a : A) : ψ a = ψ.toMonoidHom (ofAdd a) :=
+theorem coe_to_fun_apply [Monoid M] (ψ : AddChar A M) (a : A) : ψ a = ψ.toMonoidHom (ofAdd a) :=
   rfl
 #align add_char.coe_to_fun_apply AddChar.coe_to_fun_apply
 
 -- Porting note: added
-theorem mul_apply (ψ φ : AddChar A M) (a : A) : (ψ * φ) a = ψ a * φ a :=
+theorem mul_apply [CommMonoid M] (ψ φ : AddChar A M) (a : A) : (ψ * φ) a = ψ a * φ a :=
   rfl
 
 -- Porting note: added
 @[simp]
-theorem one_apply (a : A) : (1 : AddChar A M) a = 1 := rfl
+theorem one_apply [CommMonoid M] (a : A) : (1 : AddChar A M) a = 1 := rfl
 
 -- this instance was a bad idea and conflicted with `instFunLike` above
 #noalign add_char.monoid_hom_class
 
 -- Porting note(https://github.com/leanprover-community/mathlib4/issues/5229): added.
 @[ext]
-theorem ext (f g : AddChar A M) (h : ∀ x : A, f x = g x) : f = g :=
+theorem ext [Monoid M] (f g : AddChar A M) (h : ∀ x : A, f x = g x) : f = g :=
   MonoidHom.ext h
 
 /-- An additive character maps `0` to `1`. -/
 @[simp]
-theorem map_zero_one (ψ : AddChar A M) : ψ 0 = 1 := by rw [coe_to_fun_apply, ofAdd_zero, map_one]
+theorem map_zero_one [Monoid M] (ψ : AddChar A M) : ψ 0 = 1 := by
```

Comment:
```suggestion
theorem map_zero_eq_one [Monoid M] (ψ : AddChar A M) : ψ 0 = 1 := by
```

Reply:
I agree but it's out of scope for this PR

Reply:
Well, is it a good reason for not letting things improve little by little? Of course @loefflerd will do what he wants, but I can suggest this modification, no?

Reply:
But then for consistency, you also have to change to `map_add_eq_mul`  and `map_nsmul_eq_pow` etc.
The question is whether the `eq` part serves a purpose. So far nobody seems to have been confused by its missing. (And it's `map_zero`,  `map_add` etc., not `map_zero_eq_zero` and so on.)

Reply:
Well actually Filipo and Michael I was confused by the lack of it and made aliases for the correctly-named lemmas in LeanAPAP, but this is out of scope for the current PR.

Reply:
I am still sceptical of the need of breaking a PR into two just because changing one name is "out of scope". If a significant part of a PR is about changing names or docs I would agree, but perhaps we should leave room for a bit of flexibility.

Reply:
Dear Filippo, I am grateful for your suggestion, but since it is orthogonal to the original goal of the PR, I feel that as PR author I should have the option to choose not to pursue it. If you want to open a PR if your own to rename these lemmas then please go ahead